### PR TITLE
[LIVE-12806] Feature - Add Scroll & Scroll Sepolia currencies

### PR DESCRIPTION
**Pull request by lambertkevin**
### ✅ Checklist

- [x] `npx changeset` was attached.
- [x] **Covered by automatic tests.** 
- [x] **Impact of the changes:** 
  - no impact

### 📝 Description

Adding Scroll & Scroll Sepolia in the Live.
~~Waiting for tokens in CAL.~~

### ❓ Context

- **JIRA or GitHub link**: https://ledgerhq.atlassian.net/browse/LIVE-12806

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

### CODE REVIEW

File: libs/coin-modules/coin-evm/src/__tests__/unit/logic.unit.test.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import BigNumber from "bignumber.js";
 import { getEnv, setEnv } from "@ledgerhq/live-env";
+import * as EVM_TOOLS from "@ledgerhq/evm-tools/lib/message/EIP712/index";
```
Comment (lvndry) [inline libs/coin-modules/coin-evm/src/__tests__/unit/logic.unit.test.ts:3]:
```suggestion
import * as EVM_TOOLS from "@ledgerhq/evm-tools/message/EIP712/index";
```

File: libs/coin-modules/coin-evm/src/api/node/ledger.ts [hunk 2]
```diff
@@ -395,6 +394,72 @@ export const getOptimismAdditionalFees: NodeApi["getOptimismAdditionalFees"] = a
...
+  transaction,
+) => {
+  const config = getCoinConfig(currency).info;
+  const { node } = config || /* istanbul ignore next */ {};
+  if (!isLedgerNodeConfig(node)) {
+    throw new LedgerNodeUsedIncorrectly();
+  }
+
+  if (!["scroll", "scroll_sepolia"].includes(currency.id)) {
+    return new BigNumber(0);
```
Comment (lvndry) [inline libs/coin-modules/coin-evm/src/api/node/ledger.ts:417]: Shouldn't it throw an error if we can call this function from an unsupported currency ?

File: libs/coin-modules/coin-evm/src/synchronization.ts [hunk 3]
```diff
@@ -67,7 +67,7 @@ export const getAccountShape: GetAccountShape<Account> = async (infos, { blackli
           Math.max(latestSyncedHeight - SAFE_REORG_THRESHOLD, 0),
           blockHeight,
         );
-      } catch (e) {
+      } catch (e) /* istanbul ignore next: just logs */ {
```
Comment (lvndry) [inline libs/coin-modules/coin-evm/src/synchronization.ts:70]: why istanbul comments everywhere ?

File: apps/ledger-live-desktop/src/renderer/families/evm/EditTransaction/Body.tsx [hunk 4]
```diff
@@ -288,7 +289,7 @@ const Body = ({
     editType,
     transaction,
     transactionToUpdate,
-    status,
+    status: status as TransactionStatus,
```
Comment (lvndry) [inline apps/ledger-live-desktop/src/renderer/families/evm/EditTransaction/Body.tsx:292]: `as` don't seem necessary ?

File: libs/coin-modules/coin-evm/src/synchronization.ts (hunk 3 above)
Comment (lambertkevin) [reply libs/coin-modules/coin-evm/src/synchronization.ts:70]: Because that's parts of the code that we don't want/need to test, so it's unnecessary in the code coverage to have them as uncovered and potentially showing files as yellow or red as is if it was problematic.
TLDR: Nicer statistics

File: apps/ledger-live-desktop/src/renderer/families/evm/EditTransaction/Body.tsx (hunk 4 above)
Comment (lambertkevin) [reply apps/ledger-live-desktop/src/renderer/families/evm/EditTransaction/Body.tsx:292]: It is because the `TransactionStatus` from the props is generic while this is the `TransactionStatus` of coin-evm only. 
I tried to change it at the bridge level to have better typing with generics & without casting, but it's, again, a PR of hundreds of files... So it's for another time :p

File: libs/coin-modules/coin-evm/src/api/node/ledger.ts (hunk 2 above)
Comment (lambertkevin) [reply libs/coin-modules/coin-evm/src/api/node/ledger.ts:417]: We could, I just prefer a graceful fallback when possible to avoid having huge errors during development for a function called during the `prepareTransaction` phase.
